skops/hub_utils/_hf_hub.py

```diff
@@ -195,10 +202,19 @@ def _create_config(
     def recursively_default_dict() -> MutableMapping:
         return collections.defaultdict(recursively_default_dict)
 
+    if save_format == "auto":
+        extension = str(model_path).split(".")[-1]
+        if extension in ["pkl", "pickle", "joblib"]:
+            save_format = "pickle"
+        elif extension == "skops":
+            save_format = "skops"
+        else:
+            raise UserWarning("File format should be either skops or pickle!")
```

Comment:
We need a test for this (I don't think one can raise a warning ;) )

```suggestion
            raise ValueError("Cannot determine the input file format. Please indicate the format using the `model_format` argument.")
```